packages/fx-core/src/component/driver/middleware/addSpecialScriptTelemetry.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HookContext, Middleware, NextFunction } from "@feathersjs/hooks";
+import { WrapDriverContext } from "../util/wrapUtil";
+import {
+  TeamsFxTelemetryConfig,
+  TeamsFxTelemetryReporter,
+} from "../../utils/teamsFxTelemetryReporter";
+import { TelemetryConstant } from "../../constant/commonConstant";
+import { performance } from "perf_hooks";
+import { TelemetryConstants } from "../../constants";
+import { isExecutionResult } from "./addStartAndEndTelemetry";
+import { maskSecretValues } from "../../utils/envUtil";
+
+/**
+ * A special telemetry middleware for script driver running for a SWA deploy.
+ * @param eventName The event name to be sent.
+ */
+export function addSpecialScriptTelemetry(eventName: string): Middleware {
```

Comment:
Since this function is for catching swa scripts only, how about "addSWAScriptTelemetry"? "addSpecialScriptTelemetry" seems general.

Reply:
Yes, it's general as you can add any telemetry by making slight modifications to the code. However, you are correct; perhaps we should rename it back to `addSpecialScriptTelemetry` when another telemetry code is added.